lib/ash_admin/components/resource/form.ex

```diff
@@ -737,6 +739,12 @@ defmodule AshAdmin.Components.Resource.Form do
         _
       )
       when type in [Ash.Type.CiString, Ash.Type.String, Ash.Type.UUID, Ash.Type.Atom] do
+    attribute =
+      case form.source.type do
```

Comment:
I think `related_resource` could be its own variable here instead of adding a key to `attribute`.